library/python/src/maps.cc

```diff
@@ -0,0 +1,272 @@
+#include <wavemap/core/utils/query/map_interpolator.h>
+#include <wavemap/core/utils/query/query_accelerator.h>
+#include <wavemap/io/file_conversions.h>
+
+using namespace nb::literals;  // NOLINT
+
+namespace wavemap {
+void add_map_bindings(nb::module_& m) {
+  nb::class_<MapBase>(m, "Map", "Base class for wavemap maps.")
+      .def_prop_ro("empty", &MapBase::empty, "Whether the map is empty.")
+      .def_prop_ro("size", &MapBase::size,
+                   "The number of cells or nodes in the map, for fixed or "
+                   "multi-resolution maps, respectively.")
+      .def("threshold", &MapBase::threshold,
+           "Threshold the occupancy values of all cells in the map to stay "
+           "within the range specified by its min_log_odds and max_log_odds.")
+      .def("prune", &MapBase::prune,
+           "Free up memory by pruning nodes that are no longer needed. Note "
+           "that this pruning operation is lossless and does not alter the "
+           "estimated occupancy posterior.")
+      .def("pruneSmart", &MapBase::pruneSmart,
+           "Similar to prune(), but avoids de-allocating nodes that were "
+           "recently updated and will likely be used again in the near future.")
+      .def("clear", &MapBase::clear, "Erase all cells in the map.")
+      .def_prop_ro("min_cell_width", &MapBase::getMinCellWidth,
+                   "Maximum map resolution, set as width of smallest cell it "
+                   "can represent.")
+      .def_prop_ro("min_log_odds", &MapBase::getMinLogOdds,
+                   "Lower threshold for the occupancy values stored in the "
+                   "map, in log-odds.")
+      .def_prop_ro("max_log_odds", &MapBase::getMaxLogOdds,
+                   "Upper threshold for the occupancy values stored in the "
+                   "map, in log-odds.")
+      .def_prop_ro("memory_usage", &MapBase::getMemoryUsage,
+                   "The amount of memory used by the map, in bytes.")
+      .def_prop_ro("tree_height", &MapBase::getTreeHeight,
+                   "Height of the octree used to store the map. Note that this "
+                   "value is only defined for multi-resolution maps.")
+      .def_prop_ro("min_index", &MapBase::getMinIndex,
+                   "Index of the minimum corner of the map's Axis Aligned "
+                   "Bounding Box.")
+      .def_prop_ro("max_index", &MapBase::getMaxIndex,
+                   "Index of the maximum corner of the map's Axis Aligned "
+                   "Bounding Box.")
+      .def("getCellValue", &MapBase::getCellValue, "index"_a,
+           "Query the value of the map at a given index.")
+      .def("setCellValue", &MapBase::setCellValue, "index"_a,
+           "new_value"_a
+           "Set the value of the map at a given index.")
+      .def("addToCellValue", &MapBase::addToCellValue, "index"_a, "update"_a,
+           "Increment the value of the map at a given index.")
+      .def(
```

Comment:
A possible different interface would be `occupancy(coords, interpolation_enum)`. That way you don't have the two calls and it's obvious we will return occupancy data. Obviously requires a new enum but I think that may be better. Instead of an enum a string is also often used, like for example in numpy or scikit to define method params. This is an example https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html where there aren't `LogisticRegressionL2`, `LogisticRegressionLasso` etc. but simply a `penalty` param.

Reply:
It's now implemented with an `InterpolationMode.NEAREST | TRILINEAR` enum. I suggest keeping the name of the method to `map.interpolate(coords, interpolation_mode)` though, for consistency with the `.get_cell_value(...)` methods and so that the name still makes sense if people use the map to store something else (e.g. an ESDF). Once we add support for additional data channels, we could let the user choose the channel with an optional input argument (e.g. occupancy, esdf, semantic_class,...).

Reply:
The official naming convention for enums in Python seems to be ALL_CAPS. Doesn't look great though. Do you suggest sticking to that, or using something like snake_case or PascalCase?

Reply:
Makes sense, and I think there going with PascalCase would be fine too.

Reply:
Ok, I'll switch to PascalCase as it's slightly more consistent with the style we use in the C++ API and easier to read imo.